Atomic/atomic.py

```diff
@@ -435,7 +435,33 @@ def scan(self):
                 fetch = 1
             else:
                 fetch = 0
-            scan_return = json.loads(oscap_i.scan_list(scan_list, 4, fetch, timeout=99999))
+
+            json_result = ""
+            try:
+                token = oscap_i.CVEScanListAsync(
+                    scan_list, 4, fetch
+                )
+                try:
+                    while True:
+                        success, json_result = oscap_i.GetCVEScanListAsyncResults(token)
+                        if success:
+                            break
+                        time.sleep(1)
+                except:
+                    oscap_i.CancelCVEScanListAsync(token)
+                    raise
```

Comment:
Should there be a real exception raised here like ValueError with a real message?